tests/integration/src/test/java/org/mule/test/routing/ForEachSessionVarTestCase.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.test.routing;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import org.mule.api.MuleEvent;
+import org.mule.api.MuleException;
+import org.mule.api.processor.MessageProcessor;
+import org.mule.api.transport.PropertyScope;
+import org.mule.tck.junit4.FunctionalTestCase;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import org.junit.Test;
+
+public class ForEachSessionVarTestCase extends FunctionalTestCase
+{
+
+    protected static MuleEvent event;
+    private static final String MY_SESSION_LIST = "mySessionList";
+
+    @Override
+    protected String getConfigFile()
+    {
+        return "foreach-session-var-config.xml";
+    }
+
+    @Test
+    public void testSessionVars() throws Exception
+    {
+        Collection<String> expectedArray = new ArrayList<>();
+        expectedArray.add("Hello World A");
+        expectedArray.add("Hello World B");
+        muleContext.getClient().send("vm://in", getTestMuleMessage());
+        assertThat(event.getSession().<Collection<String>>getProperty(MY_SESSION_LIST), is(expectedArray));
+        assertThat(event.getMessage().<Collection<String>>getProperty(MY_SESSION_LIST, PropertyScope.SESSION), is(expectedArray));
+    }
+
```

Comment:
Add a test to ensure that event.getSession.getProperty(X) and event.getMessage.getProperty(X, PropertyScope.SESSION) point to the same map.